src/store/getters.ts

```diff
@@ -699,19 +697,20 @@ export const getEntityTypeOptions = (state: StateIF): Array<EntityI> => {
   })
 
   let options: SelectOptionsI[]
-  // special case for amalgamation - location may be CA or IN
-  if (isAmalgamation) options = [...bcOptions]
+  // special case for amalgamation - ignore location
+  if (isAmalgamation(state)) options = [...bcOptions]
   else if (isLocationBC(state)) options = [...bcOptions]
   else options = [...xproOptions]
   let n = 4
 
-  if (getEntityTypeAddToSelect(state)) {
-    getEntityTypeAddToSelect(state).rank = 4
-    options = options.concat(getEntityTypeAddToSelect(state))
+  // add recently-used entry to list
+  const entityTypeAddToSelect = getEntityTypeAddToSelect(state)
+  if (entityTypeAddToSelect) {
+    options.push({ ...entityTypeAddToSelect, rank: 4 })
```

Comment:
This was a super-tricky bug to figure out!!!

The symptom was that as soon as we selected a xpro jurisdiction, this getter (getEntityTypesBC) would go in an infinite loop with itself.

The reason is that the old code, on line 709, changed an object that this getter was reactive to.

Anyway, it's now the same functionality but simpler and working.